### ISSUE TITLE
[Web] java.lang.IllegalStateException: Not found Idx for public example.todo.desktop/Foo$composable|9030762853570172587[0]

### ISSUE THREAD

**Issue by arkivanov** [JetBrains/compose-jb#774]
I'm trying to add Web support to Decompose and playing with the TodoApp example. Bumped into the following compile crash:

```
java.lang.IllegalStateException: Not found Idx for public example.todo.desktop/Foo$composable|9030762853570172587[0]
	at org.jetbrains.kotlin.backend.common.serialization.KotlinIrLinker$IrDeserializerForFile.resolveSignatureIndex(KotlinIrLinker.kt:311)
	at org.jetbrains.kotlin.backend.common.serialization.KotlinIrLinker$IrDeserializerForFile.loadTopLevelDeclarationProto(KotlinIrLinker.kt:318)
	at org.jetbrains.kotlin.backend.common.serialization.KotlinIrLinker$IrDeserializerForFile.deserializeDeclaration(KotlinIrLinker.kt:290)
	at org.jetbrains.kotlin.backend.common.serialization.KotlinIrLinker$IrDeserializerForFile$FileDeserializationState.processPendingDeclarations(KotlinIrLinker.kt:278)
	at org.jetbrains.kotlin.backend.common.serialization.KotlinIrLinker$IrDeserializerForFile.deserializeAllFileReachableTopLevel(KotlinIrLinker.kt:461)
	at org.jetbrains.kotlin.backend.common.serialization.KotlinIrLinker$BasicIrModuleDeserializer$ModuleDeserializationState.processPendingDeclarations(KotlinIrLinker.kt:103)
	at org.jetbrains.kotlin.backend.common.serialization.KotlinIrLinker$BasicIrModuleDeserializer.deserializeReachableDeclarations(KotlinIrLinker.kt:221)
	at org.jetbrains.kotlin.backend.common.serialization.KotlinIrLinker.deserializeAllReachableTopLevels(KotlinIrLinker.kt:531)
	at org.jetbrains.kotlin.backend.common.serialization.KotlinIrLinker.findDeserializedDeclarationForSymbol(KotlinIrLinker.kt:550)
	at org.jetbrains.kotlin.backend.common.serialization.KotlinIrLinker.getDeclaration(KotlinIrLinker.kt:587)
	at org.jetbrains.kotlin.ir.util.ExternalDependenciesGeneratorKt.getDeclaration(ExternalDependenciesGenerator.kt:60)
	at org.jetbrains.kotlin.ir.util.ExternalDependenciesGenerator.generateUnboundSymbolsAsDependencies(ExternalDependenciesGenerator.kt:50)
	at org.jetbrains.kotlin.ir.backend.js.KlibKt.loadIr(klib.kt:292)
	at org.jetbrains.kotlin.ir.backend.js.CompilerKt.compile(compiler.kt:62)
	at org.jetbrains.kotlin.ir.backend.js.CompilerKt.compile$default(compiler.kt:38)
	at org.jetbrains.kotlin.cli.js.K2JsIrCompiler.doExecute(K2JsIrCompiler.kt:257)
	at org.jetbrains.kotlin.cli.js.K2JSCompiler.doExecute(K2JSCompiler.java:181)
	at org.jetbrains.kotlin.cli.js.K2JSCompiler.doExecute(K2JSCompiler.java:74)
	at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:88)
	at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:44)
	at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:98)
	at org.jetbrains.kotlin.daemon.CompileServiceImpl.compile(CompileServiceImpl.kt:1575)
	at jdk.internal.reflect.GeneratedMethodAccessor100.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at java.rmi/sun.rmi.server.UnicastServerRef.dispatch(UnicastServerRef.java:359)
	at java.rmi/sun.rmi.transport.Transport$1.run(Transport.java:200)
	at java.rmi/sun.rmi.transport.Transport$1.run(Transport.java:197)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.rmi/sun.rmi.transport.Transport.serviceCall(Transport.java:196)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport.handleMessages(TCPTransport.java:562)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run0(TCPTransport.java:796)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.lambda$run$0(TCPTransport.java:677)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run(TCPTransport.java:676)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

Here is the minimal reproducer:

```kotlin
fun main() {
    val rootElement = document.getElementById("root") as HTMLElement

    renderComposable(root = rootElement) {
        Style(Styles)

        Foo<String> {}
    }
}

class RouterState<C>

@Composable
fun <C : Any> Foo(block: @Composable (RouterState<C>) -> Unit = {}) {
}
```

**Comment by eymar** [comment]
Hi Arkadii,
I noticed this ticket only today. 

We noticed similar issues too. It's probably caused by the default value for the `block`.
Unfortunately, we didn't dig into this problem yet.
The easiest workaround for us was to not use the default value in such cases. (the default value fails when the lambda contains type parameters - <C> in your case).
Of course this is a temporary workaround and we'll get to it later.

Could you please try it without default value and let us know? 

Thanks!

**Comment by arkivanov** [comment]
Indeed, removing the default value fixes the issue. Removing `@Composable` from all parts of the `Foo` function also fixes the issue. So it looks like a bug in the Compose compiler. I will wait for a fix, not urgent for me at the moment. Thanks!

**Comment by eymar** [comment]
fixed in 1.0.0
